src/terminal/parser/stateMachine.cpp

```diff
@@ -1497,24 +1517,16 @@ void StateMachine::_EventDcsEntry(const wchar_t wch)
 
 // Routine Description:
 // - Processes a character event into an Action that occurs while in the DcsIgnore state.
-//   Events in this state will:
-//   1. Enter ground on a String terminator
-//   2. Ignore everything else.
+//   In this state the entire DCS string is considered invalid and we will ignore everything.
+//   The termination state is handled outside when an ESC is seen.
 // Arguments:
 // - wch - Character that triggered the event
 // Return Value:
 // - <none>
-void StateMachine::_EventDcsIgnore(const wchar_t wch) noexcept
+void StateMachine::_EventDcsIgnore() noexcept

```

Comment:
It’s a little weird to have just one case in this, isn’t it. But this is actually the correct way to do this.